test/scrollable_clean_calendar_test.dart

```diff
@@ -157,4 +161,53 @@ void main() {
     var postDateOfMonth = find.byKey(ValueKey('20-05-2020'));
     expect(postDateOfMonth, findsOneWidget);
   });
+
+  testWidgets(
+      'When the same date is clicked twice it should return it in both dates',
+      (WidgetTester tester) async {
+    final dateFormat = DateFormat('dd-MM-yyyy');
+
+    await tester.pumpWidget(
+        buildScrollableCleanCalendar(onRangeSelected: (firstDate, secondDate) {
+      expect(dateFormat.format(firstDate), '05-01-2020');
+      if (secondDate != null) {
+        expect(dateFormat.format(secondDate), '05-01-2020');
+      }
+    }));
+
+    await tester.tap(find.byKey(ValueKey('05-01-2020')));
+    await tester.tap(find.byKey(ValueKey('05-01-2020')));
+  });
+
+  testWidgets(
+      'When initialDateSelected is not null, the calendar should open with the date already selected',
+      (WidgetTester tester) async {
+    final dateFormat = DateFormat('dd-MM-yyyy');
+
+    await tester.pumpWidget(buildScrollableCleanCalendar(
+      locale: 'pt',
+      minDate: DateTime(2020, 2, 15),
+      maxDate: DateTime(2020, 3, 15),
+      initialDateSelected: DateTime(2020, 2, 20),
```

Comment:
If only to use the pumpWidget your test will render widget but not validate if the initial date is really checked. I think that you can validate after render if the widget is colored or rounded differently from the other dates.

What do you think?

Reply:
Okay, I think this approach is better, i will do it that way. Ty. 👍